writer.go

```diff
@@ -335,6 +338,55 @@ func NewMediaPlaylist(winsize uint, capacity uint) (*MediaPlaylist, error) {
 	return p, nil
 }
 
+// InsertSegments allows the insertion of one or multiple MediaSegments into the MediaPlaylist.
+// seqID is the sequence ID at which the new segments should be inserted to
+func (p *MediaPlaylist) InsertSegments(segments []MediaSegment, seqID uint64) error {
+	if len(segments) == 0 {
+		return ErrPlaylistEmpty
+	}
+
+	// Create copies of the segments to be inserted
```

Comment:
This is necessary because the function itself accepts this parameter by value (and not by reference)?

Reply:
Yes, I've actually done by reference first, I've changed it to Daniil's suggestion:

 append(segments, p.Segments[insertIndex:]...) will sometimes be modifying the initial segments slice (when there's enough capacity and all elements can fit in this slice) and sometimes won't (when there's not enough capacity), so if you don't want to modify the source slice, it's better to make a copy first)